include/zephyr/drivers/flash.h

```diff
@@ -57,6 +57,7 @@ struct flash_pages_layout {
 struct flash_parameters {
 	const size_t write_block_size;
 	uint8_t erase_value; /* Byte value of erased flash */
+	bool no_erase;
```

Comment:
The The https://github.com/zephyrproject-rtos/zephyr/pull/67687 defines this currently adds `characteristics.program_erase` to serve this purpose. I understand that this change has been provided here to make it possible to work while support for `no-erase` devices is not ready yet, but, once the 67687 settles and merges, the change here will have to be removed.

Reply:
@de-nordic yes this is why this change is part of a separate commit tagged with "do_not_review"

Reply:
Sorry, I looked at entire diff so have not seen the commit messages.